api/server/sdk/cloud_backup.go

```diff
@@ -0,0 +1,252 @@
+		NodeID:            req.GetNodeId(),
+	})
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to restore backup: %v", err)
+	}
+
+	return &api.SdkCloudBackupRestoreResponse{
+		RestoreVolumeId: r.RestoreVolumeID,
+	}, nil
+
+}
+
+// Delete deletes a backup
+func (s *CloudBackupServer) Delete(
+	ctx context.Context,
+	req *api.SdkCloudBackupDeleteRequest,
+) (*api.SdkCloudBackupDeleteResponse, error) {
+
+	if len(req.GetBackupId()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must provide backup id")
+	} else if len(req.GetCredentialUuid()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must provide credential uuid")
+	}
+
+	if err := s.driver.CloudBackupDelete(&api.CloudBackupDeleteRequest{
+		ID:             req.GetBackupId(),
+		CredentialUUID: req.GetCredentialUuid(),
+		Force:          req.GetForce(),
+	}); err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to delete backup: %v", err)
+	}
+
+	return &api.SdkCloudBackupDeleteResponse{}, nil
+}
+
+// DeleteAll deletes all backups for a certain volume
+func (s *CloudBackupServer) DeleteAll(
+	ctx context.Context,
+	req *api.SdkCloudBackupDeleteAllRequest,
+) (*api.SdkCloudBackupDeleteAllResponse, error) {
+
+	if len(req.GetSrcVolumeId()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must provide source volume id")
+	} else if len(req.GetCredentialUuid()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must provide credential uuid")
+	}
+
+	if err := s.driver.CloudBackupDeleteAll(&api.CloudBackupDeleteAllRequest{
+		CloudBackupGenericRequest: api.CloudBackupGenericRequest{
+			SrcVolumeID:    req.GetSrcVolumeId(),
+			CredentialUUID: req.GetCredentialUuid(),
+		},
+	}); err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to delete backup: %v", err)
+	}
+
+	return &api.SdkCloudBackupDeleteAllResponse{}, nil
+}
+
+// Enumerate returns information about the backups
+func (s *CloudBackupServer) Enumerate(
+	ctx context.Context,
+	req *api.SdkCloudBackupEnumerateRequest,
+) (*api.SdkCloudBackupEnumerateResponse, error) {
+
+	if len(req.GetCredentialUuid()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must provide credential uuid")
+	}
+
+	r, err := s.driver.CloudBackupEnumerate(&api.CloudBackupEnumerateRequest{
+		CloudBackupGenericRequest: api.CloudBackupGenericRequest{
+			SrcVolumeID:    req.GetSrcVolumeId(),
+			ClusterID:      req.GetClusterId(),
+			CredentialUUID: req.GetCredentialUuid(),
+			All:            req.GetAll(),
+		},
+	})
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to enumerate backups: %v", err)
+	}
+
+	return r.ToSdkCloudBackupEnumerateResponse(), nil
+}
+
+// Status provides status on a backup
+func (s *CloudBackupServer) Status(
+	ctx context.Context,
+	req *api.SdkCloudBackupStatusRequest,
+) (*api.SdkCloudBackupStatusResponse, error) {
+
+	r, err := s.driver.CloudBackupStatus(&api.CloudBackupStatusRequest{
```

Comment:
check for request params local and volumeId

Reply:
The object comments say that the srcvolumeid is optional. But those may be incorrect. I will check with @talakad .

Reply:
srcvolumeid is optional. if not mentioned, it will return status of all instead of requested src volumeid.